File: src/main/java/be4rjp/sclat/manager/PlayerReturnManager.java
package be4rjp.sclat.manager;

import be4rjp.sclat.VariablesKt;
import be4rjp.sclat.api.player.PlayerReturn;
import java.util.ArrayList;
import java.util.List;
import org.bukkit.scheduler.BukkitRunnable;

public class PlayerReturnManager {

	public static List<PlayerReturn> list = new ArrayList<>();

	public static boolean isReturned(String uuid) {
		for (PlayerReturn pr : list) {
			if (pr.getUUID().equals(uuid)) {
				list.remove(pr);
				return true;
			}
		}
		return false;
	}

	public static void runRemoveTask() {
		BukkitRunnable task = new BukkitRunnable() {
			@Override
			public void run() {
				try {
					list.removeIf(pr -> !pr.getFlag());
				} catch (Exception e) {
				}
			}
		};
		task.runTaskTimer(VariablesKt.getPlugin(), 0, 200);
	}

	public static void addPlayerReturn(String uuid) {
		PlayerReturn pr = new PlayerReturn(uuid);
		list.add(pr);
	}
}

File: src/main/kotlin/be4rjp/sclat/manager/PlayerReturnManager.kt
package be4rjp.sclat.manager

import be4rjp.sclat.api.player.PlayerReturn
import com.github.shynixn.mccoroutine.bukkit.launch
import kotlinx.coroutines.time.delay
import org.bukkit.plugin.java.JavaPlugin
import java.time.Duration
import java.util.UUID

object PlayerReturnManager {
    private val playerReturnMap: MutableMap<UUID, PlayerReturn> = mutableMapOf()

    fun isReturned(uuid: UUID): Boolean = playerReturnMap.remove(uuid)?.flag ?: false

    fun runRemoveTask(plugin: JavaPlugin) {
        plugin.launch {
            delay(Duration.ofSeconds(2))
            playerReturnMap.entries.removeIf { (_, pr) -> pr.flag }
        }
    }

    fun addPlayerReturn(uuid: UUID) {
        playerReturnMap[uuid] = PlayerReturn(uuid.toString())
    }
}
